feathr_project/test/test_azure_spark_e2e.py

```diff
@@ -201,3 +202,73 @@ def test_feathr_get_offline_features():
         # download result and just assert the returned result is not empty
         res_df = get_result_df(client)
         assert res_df.shape[0] > 0
+
+@pytest.mark.skip(reason="We don't have SQL server configured in CI pipeline")
+def test_feathr_get_offline_features_to_sql():
+    """
+    Test get_offline_features() can save data to SQL.
+    """
+    runner = CliRunner()
+    with runner.isolated_filesystem():
+        # runner.invoke(init, [])
+        test_workspace_dir = Path(
+            __file__).parent.resolve() / "test_user_workspace"
+        client: FeathrClient = basic_test_setup(os.path.join(test_workspace_dir, "feathr_config.yaml"))
+
+        location_id = TypedKey(key_column="DOLocationID",
+                               key_column_type=ValueType.INT32,
+                               description="location id in NYC",
+                               full_name="nyc_taxi.location_id")
+
+        feature_query = FeatureQuery(
+            feature_list=["f_location_avg_fare"], key=location_id)
+        settings = ObservationSettings(
+            observation_path="wasbs://public@azurefeathrstorage.blob.core.windows.net/sample_data/green_tripdata_2020-04.csv",
+            event_timestamp_column="lpep_dropoff_datetime",
+            timestamp_format="yyyy-MM-dd HH:mm:ss")
+
+        now = datetime.now()
+        # Set DB user and password before submitting job
+        # os.environ[f"o_USER"] = "some_user@feathrtestsql4"
+        # os.environ[f"o_PASSWORD"] = "some_password"
+        output_path = JdbcSink(name="o",
+                               url="jdbc:sqlserver://feathrtestsql4.database.windows.net:1433;database=testsql;encrypt=true;",
+                               dbtable=''.join(['feathrazure_cijob','_', str(now.minute), '_', str(now.second)]),
+                               auth="USERPASS")
+
+        client.get_offline_features(observation_settings=settings,
+                                    feature_query=feature_query,
+                                    output_path=output_path)
+
+        # assuming the job can successfully run; otherwise it will throw exception
+        client.wait_job_to_finish(timeout_sec=Constants.SPARK_JOB_TIMEOUT_SECONDS)
+
+        # We may not have SQL client package installed so skip the verification
+        # res_df = get_result_df(client)
+        # assert res_df.shape[0] > 0
+
+@pytest.mark.skip(reason="We don't have CosmosDb configured in CI pipeline")
```

Comment:
Let's make sure we configure those resources in the test environment and make sure those test can run successfully

Reply:
Let's talk about it today. We should have the cloud test setup rather than delaying it.

Reply:
BTW it should only take less than 1 hour to setup the cloud resource and make sure all the testing can pass (if they have been verified)